### PR TITLE
Vite build

### PR DESCRIPTION
**Pull request by 4350pChris**
Switch to vite in order to build vue components (needed for #59)

### CODE REVIEW

**Comment by brillout** [comment]
> Error: [vike][Wrong Usage] The import path 'vike-vue' in /pages/+config.h.ts couldn't be resolved: does 'vike-vue' exist?

`package.json#exports["."]["require"]` is needed (because Vike uses `require.resolve()` which unfortunately doesn't have an ESM counterpart).

**Comment by 4350pChris** [comment]
Ah, thanks. Should be able to build now.

File: vike-vue/package.json [hunk 1]
```diff
@@ -2,33 +2,56 @@
   "name": "vike-vue",
   "version": "0.5.3",
   "type": "module",
-  "main": "./dist/renderer/+config.js",
+  "main": "./dist/+config.js",
```
Comment (AurelienLourot) [inline vike-vue/package.json:5]: Is there a specific reason for this change? I think in Vike's docs we advertize placing config files either under `/renderer/` or somewhere under `/pages/`.

**Review by AurelienLourot — COMMENTED** [review]
Sorry for joining the party late :) Thank you for doing this, LGTM. Just one question.

File: vike-vue/package.json (hunk 1 above)
Comment (brillout) [reply vike-vue/package.json:5]: We just discussed this, it only affects `dist/` so it's fine.

**Review by AurelienLourot — APPROVED** [review]

**Comment by brillout** [comment]
Do we need `tsconfig.node.json`? Is there a reason for not using only one `tsconfig.json`?

**Comment by brillout** [comment]
FYI, we don't need the `.cjs` in `dist/`. I think the `require` can point to ESM files because Vike only needs `require` for its `require.resolve()` call, Vike then uses `import()`. To make things less confusing we can use `default` for replacing both `import` and `require`.

**Comment by brillout** [comment]
I removed `vue-tsc`, it seems like we don't need it?

File: vike-vue/package.json (hunk 1 above)
Comment (4350pChris) [reply vike-vue/package.json]: Honestly I just didn't know how to make vite keep the folder structure. It places all js files at the top right now and the type files in the folders. I just couldn't find the right config to place the js files adjacent.

**Comment by brillout** [comment]
I'm looking at how to externalize all vike imports (how we currently do it is quite error prone), but so far I couldn't find a better alternative.

**Comment by brillout** [comment]
Hm, there doesn't seem to be any alternative. That's quite a bummer.

This PR LGTM now.

@4350pChris I simplified the PR to the bare minimum, but let me know if you see any issues.

**Review by AurelienLourot — APPROVED** [review]
LGTM :+1:

**Comment by brillout** [comment]
I found a workaround. It was worth it... we were already missing `vue/server-renderer`.

**Review by AurelienLourot — APPROVED** [review]

**Review by AurelienLourot — APPROVED** [review]

**Comment by AurelienLourot** [comment]
@4350pChris WDYT?

**Comment by 4350pChris** [comment]
Thanks for cleaning this up @brillout 
LGTM

**Pull request merged**